framework/src/variables/MooseVariableFV.C

```diff
@@ -0,0 +1,354 @@
+{
+  return MooseVariableField<OutputType>::validParams();
+}
+
+template <typename OutputType>
+MooseVariableFV<OutputType>::MooseVariableFV(const InputParameters & parameters)
+  : MooseVariableField<OutputType>(parameters)
+{
+  _element_data = libmesh_make_unique<MooseVariableDataFV<OutputType>>(
+      *this, _sys, _tid, Moose::ElementType::Element, this->_assembly.elem());
+  _neighbor_data = libmesh_make_unique<MooseVariableDataFV<OutputType>>(
+      *this, _sys, _tid, Moose::ElementType::Neighbor, this->_assembly.neighbor());
+}
+
+template <typename OutputType>
+const std::set<SubdomainID> &
+MooseVariableFV<OutputType>::activeSubdomains() const
+{
+  return this->_sys.system().variable(_var_num).active_subdomains();
+}
+
+template <typename OutputType>
+Moose::VarFieldType
+MooseVariableFV<OutputType>::fieldType() const
+{
+  if (std::is_same<OutputType, Real>::value)
+    return Moose::VarFieldType::VAR_FIELD_STANDARD;
+  else if (std::is_same<OutputType, RealVectorValue>::value)
+    return Moose::VarFieldType::VAR_FIELD_VECTOR;
+  else if (std::is_same<OutputType, RealEigenVector>::value)
+    return Moose::VarFieldType::VAR_FIELD_ARRAY;
+  else
+    mooseError("Unknown variable field type");
+}
+
+template <typename OutputType>
+bool
+MooseVariableFV<OutputType>::activeOnSubdomain(SubdomainID subdomain) const
+{
+  return this->_sys.system().variable(_var_num).active_on_subdomain(subdomain);
+}
+
+template <typename OutputType>
+void
+MooseVariableFV<OutputType>::clearDofIndices()
+{
+  _element_data->clearDofIndices();
+}
+
+template <typename OutputType>
+typename MooseVariableFV<OutputType>::OutputData
+MooseVariableFV<OutputType>::getElementalValue(const Elem * elem, unsigned int idx) const
+{
+  return _element_data->getElementalValue(elem, Moose::Current, idx);
+}
+
+template <typename OutputType>
+typename MooseVariableFV<OutputType>::OutputData
+MooseVariableFV<OutputType>::getElementalValueOld(const Elem * elem, unsigned int idx) const
+{
+  return _element_data->getElementalValue(elem, Moose::Old, idx);
+}
+
+template <typename OutputType>
+typename MooseVariableFV<OutputType>::OutputData
+MooseVariableFV<OutputType>::getElementalValueOlder(const Elem * elem, unsigned int idx) const
+{
+  return _element_data->getElementalValue(elem, Moose::Older, idx);
+}
+
+template <typename OutputType>
+void
+MooseVariableFV<OutputType>::insert(NumericVector<Number> & residual)
+{
+  _element_data->insert(residual);
+}
+
+template <typename OutputType>
+void
+MooseVariableFV<OutputType>::add(NumericVector<Number> & residual)
+{
+  _element_data->add(residual);
+}
+
+template <typename OutputType>
+void
+MooseVariableFV<OutputType>::addSolution(const DenseVector<Number> & v)
+{
+  _element_data->addSolution(this->_sys.solution(), v);
+}
+
+template <typename OutputType>
+void
+MooseVariableFV<OutputType>::addSolutionNeighbor(const DenseVector<Number> & v)
+{
+  _neighbor_data->addSolution(this->_sys.solution(), v);
```

Comment:
I was wondering what context `add` is introduced in several places.

Reply:
This is also not used by FV - probably should remove as well - we just use your amazing tagging interface instead.